src/webgpu/shader/types.ts

```diff
@@ -204,10 +216,34 @@ export function* generateTypes({
   containerType: ContainerType;
   /** Whether to wrap the baseType in `atomic<>`. */
   isAtomic?: boolean;
-}) {
+}): Generator<
+  {
+    type: string; // WGSL name for the generated type
+    _kTypeInfo: {
+      // WGSL name for:
+      // - the generated type if it is scalar or atomic
+      // - the column vector type if the generated type is a matrix
+      // - the base type if the generated type is an array
+      elementBaseType: string;
+      // Layout details if host-shareable, and undefined otherwise.
+      layout: undefined | AlignmentAndSize;
+      supportsAtomics: boolean;
+      // The number of elementBaseType items in the container.
+      arrayLength: number;
+      // 0 for scalar and vector.
+      // For a matrix type, this is the number of rows in the matrix.
+      innerLength?: number;
```

Comment:
nit: write these as `/** doc comments */`. Editors should be able to pick those docs up on hover on usages of these members, I think.

```suggestion
    /** WGSL name for the generated type */
    type: string;
    _kTypeInfo: {
      /**
       * WGSL name for:
       * - the generated type if it is scalar or atomic
       * - the column vector type if the generated type is a matrix
       * - the base type if the generated type is an array
       */
      elementBaseType: string;
      /** Layout details if host-shareable, and undefined otherwise. */
      layout: undefined | AlignmentAndSize;
      supportsAtomics: boolean;
      /** The number of elementBaseType items in the container. */
      arrayLength: number;
      /**
       * 0 for scalar and vector.
       * For a matrix type, this is the number of rows in the matrix.
       */
      innerLength?: number;
```